packages/layout/src/tree-layout/layout-manager/index.js

```diff
@@ -140,8 +144,22 @@ export default class LayoutManager {
         return this._layoutDef.componentMap().get(componentName);
     }
 
-    removeComponent (id) {
-        return removeElement(id);
+    removeComponent (name) {
+        const component = this.getComponent(name);
+        if (component) {
+            if (name === GRID) {
+                component.component.map(comp => comp.map((co) => {
+                    this._layoutDef.componentMap().delete(co.name());
+                    const deleteElementId = co.renderAt();
+                    return removeElement(deleteElementId);
+                }));
+            } else {
+                this._layoutDef.componentMap().delete(name);
+                const deleteElementId = component.renderAt();
+                return removeElement(deleteElementId);
+            }
+        }
+        return this;
```

Comment:
Lines number 152-154 are exactly the same as 157-159.

Also, why is the `name ===GRID` check done here? Call the grid component remove method, which will have the responsibility to remove all its subcomponents.